src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -166,5 +167,29 @@ public void TestUdfWithReturnAsTimestampType()
                 }
             }
         }
+
+        /// <summary>
+        /// Test to validate UDFs defined in separate threads work.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithMultipleThreads()
+        {
+            try
+            {
+                static void DefineUdf() => Udf<string, string>(str => str);
```

Comment:
for this local function, does it matter if it is static ? If not let's just make it non static.

Reply:
I made it static because VStudio suggested 'Local functions can be made static' but I can remove that if preferred.